### DISCUSSION TITLE
How do I modify built assemblies before they are processed by NativeAOT?

### DISCUSSION THREAD

**Discussion by slxdy** [dotnet/runtime#94940] [Q&A] [answered]
I wonder, is there any way to modify assemblies after they were compiled to IL and before NativeAOT processes them?

**Comment by slxdy** [comment]
Actually, more specifically, can I `dotnet run` another project in the solution after the main project is built?

**Comment by vitek-karas** [comment]
@sbomer - you probably know right away what is the right place in the SDK - it should be the same point we use for signing things before they go into single-file, right?

**Comment by MichalPetryka** [comment, marked as answer]
Doesn't an MSBuild Task with target set to after build work in this case?

**Reply by slxdy** [reply]
Actually, you may be right. Is there a way to `dotnet run [another/project/path]` but also pass the output path as an argument?

**Reply by slxdy** [reply]
Alright, I figured it out. I did just that by setting the post-build event to `dotnet run --project $(SolutionDir)PostProcessor -- "$(TargetPath)"`. Thanks for the help

**Comment by slxdy** [comment]
Alright, I found the solution. I simply ran my build processor on the post-build event only when the output is updated.
